time/unit.go

```diff
@@ -56,6 +56,14 @@ func (tu Unit) Value() (time.Duration, error) {
 	return 0, errUnrecognizedTimeUnit
 }
 
+// Count returns the number of units contained within the duration.
+func (tu Unit) Count(d time.Duration) (int, error) {
+	if dur, found := unitsToDuration[tu]; found {
+		return int(d / dur), nil
```

Comment:
same point as above about negative counts. Might be better to guarantee `>=0` in the contract.